extensions/wikia/UserActivity/UserActivityController.class.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace UserActivity;
+
+class Controller extends \WikiaController {
+	const DEFAULT_TEMPLATE_ENGINE = \WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+	const NUM_ARTICLES_SHOWN = 3;
+
+	public function index() {
+		$user = \F::app()->wg->User;
+		$limit = $this->getVal( 'limit' );
+		$offset = $this->getVal( 'offset' );
+		$order = $this->getVal( 'order' );
+
+		$result = [
+			'total' => 0,
+			'totalReturned' => 0,
+			'items' => []
+		];
+
+		if ( empty( $user ) ) {
+			return;
+		}
+
+		if ( $user->isBlocked() ) {
+			return;
+		}
+
+		if ( !$user->isLoggedIn() ) {
+			return;
+		}
+
+		$contribs = new \LookupContribsCore( $user->getName() );
+		if ( !$contribs->checkUser() ) {
+			$this->response->setData( $result );
+			return;
+		}
+
+		$contribs->setLimit( $limit );
+		$contribs->setOffset( $offset );
+
+		$activity = $contribs->checkUserActivity(
+			$addEditCount = true,
+			$order,
+			$limit,
+			$offset
+		);
+
+		if ( !empty( $activity ) ) {
+			$result['items'] = $this->formatItems( $activity['data'] );
+			$result['total'] = $activity['cnt'];
+			$result['totalReturned'] = count( $result['items'] );
+		}
+
+		$this->response->setData( $result );
+	}
+
+	protected function formatItems( $items ) {
+		$flattened = [];
+		foreach ( $items as $sortKey => $contribItem ) {
+			$lastEditTS = wfTimestamp( TS_MW, $contribItem[ 'last_edit' ] );
+			$localizedDate = \F::app()->wg->Lang->timeanddate( $lastEditTS, $localTZ = true );
```

Comment:
This tells the reader that I want this date in the local timezone, rather than "the second argument to `timeanddate` is true for some reason"